### DISCUSSION TITLE
Discussion of abandonment note...

### DISCUSSION THREAD

**Discussion by ylluminate** [dmarmor/epichrome#338] [General]
I wanted to create a location to have further open discussions about the project abandonment note found here:
https://github.com/dmarmor/epichrome/blob/master/README.md#important-note

@dmarmor could you please further the details behind your your statement here?:
> coming changes to macOS and the Chrome extension ecosystem.

**Comment by ylluminarious** [comment]
@dmarmor Yes, I'm also curious about what was meant by these coming changes. Does it have something to do with macOS Big Sur / the gradual move towards the new ARM-based processors? What about Chrome? Are there big changes coming there too?

**Comment by ylluminate** [comment]
FWIW, after looking more closely at @dmarmor's [options](https://github.com/dmarmor/epichrome#important-note), I've found that Coherence X is the closest option to Epichrome - and provides *perhaps* an even smoother experience. Further the developer said a major release will be coming out later this month that will improve Coherence X even further. He also said that if enough people request Epichrome migration to Coherence X then they will consider adding it so that we can take our Epichrome app config's with us...

Presently the options appear to include the same Chrome + Brave options but also has the Canary option as well.

**Comment by gregschroeder** [comment]
1. NOOOOOOOOOOOOOOO!  You can't quit!  :)
2. Thanks, @dmarmor, for all that you've done - Epichrome is great, particularly this latest major release.
3. See # 1

Coherence does seem the closest, but I haven't had a chance to test it very much.  Very limited trial before you need to buy it.

It does not seem to be scriptable, and I use this 'feature' (the direct calling of `epichrome.sh`) heavily. I also use @dmarmor's suggested trick of using a zip with a known base state as a template, then using `edit` to change that base into my desired destination.  In this way, everything is copied - extensions, bookmarks, passwords, history - which is VERY useful.  I also wrote a little script to update bookmarks, etc.

I don't see any way to do that in Coherence.  Profile copying is supported but doesn't seem to include much of the above except extensions.  I don't know how I might tackle the common-base-template problem.  There are little nits, like requiring perfectly square icons, that are solvable but annoying.

As for helping maintain Epichrome, I would be open to helping, but I was unable to do the build locally for a number of reasons the last time I tried - needed Adobe InDesign or Photoshop or something, needed to be able to codesign, etc.

**Comment by ylluminate** [comment]
The Coherence X developer asked for folks to email him at `bzg [at] bzgapps [dot] com` to request Epichrome importation and any other feature requests.

**Comment by dmarmor** [comment]
Hi all. Thanks for checking in. I've converted this to a discussion.

The main issues I see ahead are:

1. The trend with macOS is ever-tightening app security (for good reason, mainly). But with each release, it becomes more difficult to get around that security and create ad hoc, unsigned apps that will work the way users expect them to. I believe we are probably only a release or two away from a situation where users will not be able to run unsigned apps unless they take specific steps to relax their security across the board. There are (possibly) ways to deal with that, but they will require a significant rewrite of Epichrome, at least as major as the 2.2-2.3 or 2.3-2.4 changes.

1. As far as Chrome goes, they are phasing in manifest V3, which replaces background pages with service workers, and would require a complete, ground-up redesign. of the Epichrome runtime extension. I haven't looked into it enough to know whether it will even be *possible* to still do the kind of global link redirecting the extension now does, but at least it will be difficult to manage without a persistent background page.

1. Epichrome began as a single bash script for my own personal use. Because of that origin, it has evolved as an ever-growing, ever more cumbersome set of bash scripts with a frankly terrible JXA front-end. The only way I see to really take it forward (beyond the challenges listed above) would be to entirely rewrite the software in Swift, possibly with a Chrome extension for the front-end.

I hope this helps understand the really difficult decision I had to make. Any one of the issues I've listed is beyond what I will have time to do going forward, let alone all three of them.

@gregschroeder, if you're interested in working on maintaining Epichrome, please e-mail me at the e-mail listed at the bottom of the note in the Readme. Since you last tried to build, I've done some work to try to make Epichrome more able to be built on systems other than mine, but I haven't had a chance to test this. I'd be happy to work through any kinks with you.

Thanks everyone for all your support for Epichrome over the years.

**Reply by dpickett** [reply]
thank you and though I'm sad to see development ending, I totally get it!
